mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/DatasourceAttributeExtractor.scala

```diff
@@ -34,16 +40,117 @@ private[autologging] object DatasourceAttributeExtractor {
+          case l: LeafNode =>
+            Seq(l)
+          case other: LogicalPlan => getLeafNodes(other)
+        }
+      }
+    }
+  }
+
+  /**
+   * Get SparkTableInfo representing the datasource(s) that were read from a SparkListenerEvent
+   * assumed to have a QueryExecution field named "qe".
+   */
+  def getTableInfos(event: SparkListenerSQLExecutionEnd): Seq[SparkTableInfo] = {
+    val qe = SparkAutologgingUtils.getQueryExecution(event)
+    if (qe != null) {
+      val leafNodes = getLeafNodes(qe.analyzed)
+      leafNodes.flatMap(getTableInfoToLog)
+    } else {
+      Seq.empty
+    }
+  }
+}
+
+object DatasourceAttributeExtractor extends DatasourceAttributeExtractorBase {
+  // TODO: attempt to detect Delta table info when Delta Lake becomes compatible with Spark 3.0
+  override def maybeGetDeltaTableInfo(leafNode: LogicalPlan): Option[SparkTableInfo] = None
+}
+
+/** Datasource attribute extractor for REPL-ID aware environments (e.g. Databricks) */
+object ReplAwareDatasourceAttributeExtractor extends DatasourceAttributeExtractorBase {
+  override protected def maybeGetDeltaTableInfo(leafNode: LogicalPlan): Option[SparkTableInfo] = {
     leafNode match {
-      case relation: DataSourceV2Relation =>
-        getSparkTableInfoFromTable(relation.table)
-      case other =>
-        None
+      case lr: LogicalRelation =>
+        // First, check whether LogicalRelation is a Delta table
+        val obj = ReflectionUtils.getScalaObjectByName("com.databricks.sql.transaction.tahoe.DeltaTable")
+        val deltaFileIndexOpt = ReflectionUtils.callMethod(obj, "unapply", Seq(lr)).asInstanceOf[Option[Any]]
+        deltaFileIndexOpt.map(fileIndex => {
+          val path = ReflectionUtils.getField(fileIndex, "path").toString
+          val versionOpt = Option(ReflectionUtils.callMethod(fileIndex, "tableVersion", Seq.empty)).map(_.toString)
+          SparkTableInfo(path, versionOpt, Option("delta"))
+        })
+      case other => None
+    }
+  }
+
+  override def getTableInfoToLog(leafNode: LogicalPlan): Option[SparkTableInfo] = {
+    val res = super.getTableInfoToLog(leafNode)
+    if (res.isDefined) {
+      res
+    } else {
+      leafNode match {
+        case LogicalRelation(HadoopFsRelation(index, _, _, _, fileFormat, _), _, _, _) =>
+          val path: String = index.rootPaths.headOption.map(_.toString).getOrElse("unknown")
+          val formatOpt = fileFormat match {
+            case format: DataSourceRegister => Option(format.shortName)
+            case _ => None
+          }
+          Option(SparkTableInfo(path, None, formatOpt))
+        case _ => None
+      }
+    }
+  }
+
+  private def tryRedactString(value: String): String = {
+    try {
+      val redactor = ReflectionUtils.getScalaObjectByName(
+        "com.databricks.spark.util.DatabricksSparkLogRedactor")
+      ReflectionUtils.callMethod(redactor, "redact", Seq(value)).asInstanceOf[String]
+    } catch {
+      case NonFatal(e) =>
+        value
```

Comment:
why do we call try redact here? is it safe to return unredacted values? When can that happen?

Reply:
Good question, we apply redaction when available to remove e.g. S3 credentials from datasource paths. Added a clarifying comment, also seems reasonable to just fail autologging instead of logging unredacted values, so will remove the `catch NonFatal(e)`  here